ide/src/setup.ts

```diff
@@ -177,27 +177,12 @@ export async function setup(
     }
   }
 
-  const tdcp = new (class implements vscode.TextDocumentContentProvider {
-    readonly uri = vscode.Uri.parse("flowistry://build-error");
-    readonly eventEmitter = new vscode.EventEmitter<vscode.Uri>();
-    contents: string = "";
-
-    provideTextDocumentContent(
-      _uri: vscode.Uri
-    ): vscode.ProviderResult<string> {
-      return `Flowistry could not run because your project failed to build with error:\n${this.contents}`;
-    }
-
-    get onDidChange(): vscode.Event<vscode.Uri> {
-      return this.eventEmitter.event;
-    }
-  })();
-
+  const tdcp = new FlowistryBuildErrorDocument();
   context.subscriptions.push(
     vscode.workspace.registerTextDocumentContentProvider("flowistry", tdcp)
   );
 
-  return async <T>(args: string) => {
+  return async <T>(args: string, hide_error?: boolean) => {
```

Comment:
Rather than add this parameter, I think the cleaner approach is to move the error rendering out of `CallFlowistry` and into the extensions. One possibility is to have this function throw two types of exceptions to distinguish build errors vs. Flowistry errors.